src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1275,25 +1275,35 @@ internal static BoundExpression NullOrDefault(TypeSymbol typeSymbol, SyntaxNode
 #endif
             )
         {
+            MethodSymbol containingMethod = this.CurrentFunction;
```

Comment:
>this.CurrentFunction [](start = 44, length = 20)

Is this set to constructor for constructor initializer and field initializers?

Reply:
I added a test for this, you should see if it matches what you were thinking. (I also verified that, yes, the value of `containingMethod` is the constructor here).